subworkflows/uniref90_generation/main.nf

```diff
@@ -0,0 +1,44 @@
+include { UNIREF90_RHEA_FILTER                  } from '../../modules/local/uniref90_rhea_filter/main.nf'
+include { UNIREF90_NON_VIRAL_FILTER             } from '../../modules/local/uniref90_non_viral_filter/main.nf'
+include { REFORMAT_RHEA_CHEBI                   } from '../../modules/local/reformat_rhea_chebi/main.nf'
+include { DIAMOND_MAKEDB as DIAMOND_MAKEDB_RHEA } from '../../modules/nf-core/diamond/makedb/main.nf'
+include { DIAMOND_MAKEDB as DIAMOND_MAKEDB_TAXA } from '../../modules/nf-core/diamond/makedb/main.nf'
+
+workflow UNIREF90_GENERATION {
+
+    main:
+        uniref90_fasta       = file(params.uniref90_download_fasta)
+        uniprot_rhea_mapping = file(params.uniprot_rhea_mapping)
+
+        UNIREF90_RHEA_FILTER(
+            uniref90_fasta,
+            uniprot_rhea_mapping
+        )
+
+        UNIREF90_RHEA_FILTER.out.filtered_fasta
+            .map { filepath -> 
+                [[id: "uniref90_rhea_${params.uniref90_version}_${params.uniprotKB_access_date}"], filepath]
+            }
+            .set { diamond_makedb_rhea_ch }
+        DIAMOND_MAKEDB_RHEA(diamond_makedb_rhea_ch, [], [], [])
+
+        uniref90_batches_ch = Channel.fromPath(uniref90_fasta).splitFasta(by: params.uniref90_batch_size, file: 'uniref90')
```

Comment:
How big is uniref90? If it's too big we may want to have splitFasta to store the files on disk (the default way is to store the chunks in memory)

Reply:
It's quite big, ~60Gb when compressed. 
But I'm using `file` option already to create files. It doesn't use memory 🙂

Reply:
perfect.. I missed the file thingy!